test/leSpec.js

```diff
@@ -310,53 +321,155 @@ describe('catch all option', function () {
 
 describe('destroys log streams', function () {
     it('default', function () {
-        LE.init(TOKEN);
-        LE.destroy();
+        IOPS.init({
+                token: TOKEN,
+                region: 'eu'
+        });
+        IOPS.destroy();
 
         expect(function () {
-            LE.init(TOKEN);
+            IOPS.init({
+                token: TOKEN,
+                region: 'eu'
+            });
         }).not.toThrow();
     });
 
     it('custom name', function () {
-        LE.init({
+        IOPS.init({
             token: TOKEN,
-            name: 'test'
+            name: 'test',
+            region: 'eu'
         });
-        LE.destroy('test');
+        IOPS.destroy('test');
 
         expect(function () {
-            LE.init({
+            IOPS.init({
                 token: TOKEN,
-                name: 'test'
+                name: 'test',
+                region: 'eu'
             });
         }).not.toThrow();
-        LE.destroy('test');
+        IOPS.destroy('test');
     });
 
     afterEach(destroy);
 });
 
+describe('tests for SSL', function () {
+    beforeEach(mockXMLHttpRequests);
+    beforeEach(addGetJson);
+
+    it('SSL option set to true leads to \"https\"', function () {
+        IOPS.init({
+            token: TOKEN,
+            name: 'test',
+            ssl: true,
+            region: 'eu'
+        });
+        IOPS.log("Test");
+        var url = this.requestList[0].url;
+        assert(url.search('https') === 0);
+        IOPS.destroy('test');
+    });
+
+    it('SSL option set to false leads to \"http\"', function () {
+        IOPS.init({
+            token: TOKEN,
+            name: 'test',
+            ssl: false,
+            region: 'eu'
+        });
+        IOPS.log("Test");
+        var url = this.requestList[0].url;
+        assert(url.search("http") === 0);
```

Comment:
This will pass even if the url begins with `https`; you'd want to check that the url begins with `http` and also check that the url _doesn't_ begin with `https` (or just check for `http:`)

A good rule of thumb for tests like this is to make them fail first, then make them pass (i.e. break the logic so that it returns `https`, observe the test failing, then fix the logic)

Reply:
Very good point. I missed that.